src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -477,20 +477,6 @@ The URL is an identifier only. Kibana and your browser will never access this UR
 
     // `false` is a valid value
     res.mapStyle = this._config?.mapStyle === undefined ? `default` : this._config.mapStyle;
-    if (res.mapStyle !== `default` && res.mapStyle !== false) {
```

Comment:
There should still be a warning message if mapStyle does not match any available TMS service so users know what values are available and value. The comparison should ensure value is "default", false, or any of the ids from EMS when EMS is enabled.

Reply:
We already have that message in src/plugins/vis_type_vega/public/vega_view/vega_map_view.js see 

```
  const mapOptions = tmsServices.find((s) => s.id === mapStyle);

    if (!mapOptions) {
      this.onWarn(
        i18n.translate('visTypeVega.mapView.mapStyleNotFoundWarningMessage', {
          defaultMessage: '{mapStyleParam} was not found',
          values: { mapStyleParam: `"mapStyle":${mapStyle}` },
        })
      );
      return null;
    }
```

Reply:
Then how about expanding the message to include expected values?

Reply:
np 
![image](https://user-images.githubusercontent.com/20072247/104814664-47668c00-5821-11eb-8023-a3ffc383c8ed.png)